python_bindings/bindings.cpp

```diff
@@ -0,0 +1,1144 @@
+                 return initial_network_settings;
+             }),
+             "file"_a = std::optional<std::string>{},
+             "number_of_agents"_a = 200,
+             "connections_per_agent"_a = 10)
+        .def_readwrite("file", &Seldon::Config::InitialNetworkSettings::file)
+        .def_readwrite("number_of_agents", &Seldon::Config::InitialNetworkSettings::n_agents)
+        .def_readwrite("connections_per_agent", &Seldon::Config::InitialNetworkSettings::n_connections);
+
+    py::class_<Seldon::Config::SimulationOptions>(m, "SimulationOptions")
+        .def(py::init([](std::string model_string,
+                         std::size_t rng_seed,
+                         Seldon::Config::OutputSettings output_settings,
+                         std::variant<Seldon::Config::DeGrootSettings,
+                                      Seldon::Config::ActivityDrivenSettings,
+                                      Seldon::Config::ActivityDrivenInertialSettings,
+                                      Seldon::Config::DeffuantSettings> model_settings,
+                         Seldon::Config::InitialNetworkSettings network_settings) {
+                 Seldon::Config::SimulationOptions simulation_options;
+                 simulation_options.model_string = model_string;
+                 simulation_options.output_settings = output_settings;
+                 simulation_options.rng_seed = rng_seed;
+                 simulation_options.model_settings = model_settings;
+                 simulation_options.network_settings = network_settings;
+                 if (simulation_options.model_string == "DeGroot") {
+                     simulation_options.model = Seldon::Config::Model::DeGroot;
+                 } else if (simulation_options.model_string == "ActivityDriven") {
+                     simulation_options.model = Seldon::Config::Model::ActivityDrivenModel;
+                 } else if (simulation_options.model_string == "Deffuant") {
+                     simulation_options.model = Seldon::Config::Model::DeffuantModel;
+                 } else if (simulation_options.model_string == "ActivityDrivenInertial") {
+                     simulation_options.model = Seldon::Config::Model::ActivityDrivenInertial;
+                 } else {
+                     throw std::runtime_error("Invalid model string. Supported models are DeGroot, ActivityDriven, Deffuant, ActivityDrivenInertial");
+                 }
+                 return simulation_options;
+             }),
+             "model_string"_a = "DeGroot",
+             "rng_seed"_a = std::random_device()(),
+             "output_settings"_a = Seldon::Config::OutputSettings{},
+             "model_settings"_a = Seldon::Config::DeGrootSettings{},
+             "network_settings"_a = Seldon::Config::InitialNetworkSettings{})
+        .def_readwrite("model_string", &Seldon::Config::SimulationOptions::model_string)
+        .def_readwrite("rng_seed", &Seldon::Config::SimulationOptions::rng_seed)
```

Comment:
`std::random_device` returns an `unsigned int` but the readwrite will allow the user to set undefined values (e.g. negatives, floats). Consider constraining with a property and a lambda setter.

Reply:
Maybe https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution instead